### PR TITLE
xsar 0.6 pre release

### PR DESCRIPTION
**Pull request by oarcher**
Bug fixes and improvements.

The main one is the use of dask `Actor` for `Sentinel1Meta`, so when running on a cluster, the `s1meta` instance is running on one worker, and might be serialized to other workers, without loosing it's state.

This induced a memory leak, because the Actor is never released. (ref https://github.com/dask/distributed/issues/5610)

As a temporary workaround, variable cache was disabled in `xsar.xml_parser.py`. (ref #23 )

### CODE REVIEW

**Comment by oarcher** [comment]
@alevieux , checks are in error state because some dependencies are missing ( see https://github.com/conda-forge/xsar-feedstock/pull/1)

File: src/xsar/sentinel1_dataset.py [hunk 1]
```diff
@@ -91,13 +87,15 @@ def __init__(self, dataset_id, resolution=None,
...
-            xml_parser = XmlParser(
-                xpath_mappings=sentinel1_xml_mappings.xpath_mappings,
-                compounds_vars=sentinel1_xml_mappings.compounds_vars,
-                namespaces=sentinel1_xml_mappings.namespaces)
-            self.s1meta = Sentinel1Meta(dataset_id, xml_parser=xml_parser)
+            self.s1meta = BlockingActorProxy(Sentinel1Meta, dataset_id)
+            # check serializable
+            #import pickle
+            #s1meta = pickle.loads(pickle.dumps(self.s1meta))
+            #assert isinstance(tuple,s1meta.coords2ll(100, 100))
```
Comment (alevieux) [inline src/xsar/sentinel1_dataset.py:94]: Removing comment line

File: setup.py [hunk 2]
```diff
@@ -33,9 +33,10 @@
         'fsspec',
         'aiohttp',
     ],
-    entry_points={
-        "xarray.backends": ["xsar=xsar.xarray_backends:XsarXarrayBackend"]
-    },
+    # disabled while migrating from rasterio to rioxarray
+    #entry_points={
+    #    "xarray.backends": ["xsar=xsar.xarray_backends:XsarXarrayBackend"]
+    #},
```
Comment (alevieux) [inline setup.py:39]: Removing comment line

File: src/xsar/sentinel1_dataset.py [hunk 3]
```diff
@@ -126,8 +124,16 @@ def __init__(self, dataset_id, resolution=None,
...
+        # Is it a memory leak ?
+        # see https://github.com/dask/distributed/issues/5610
+        # tmp_f = self.s1meta.to_dict("all")
+        # del tmp_f
+        # return
 
+        self._dataset.attrs.update(self.s1meta.to_dict("all"))
         # load subswath geometry
         # self.geometry = self.load_geometry(self.files['noise'].iloc[0])
         # self._dataset.attrs['geometry'] = self.geometry
```
Comment (alevieux) [inline src/xsar/sentinel1_dataset.py:139]: Removing comment line

File: src/xsar/utils.py [hunk 4]
```diff
@@ -303,37 +220,21 @@ def _evaluate_from_coords(block, f, coords, block_info=None, dtype=None):
 
     from_coords = bind(_evaluate_from_coords, ..., ..., coords.values(), dtype=dtype)
 
+    #try:
+    #    # in func is a future, get it
+    #    func = func.result()
+    #except AttributeError:
+    #    pass
+
```
Comment (alevieux) [inline src/xsar/utils.py:228]: Removing comment line

File: src/xsar/utils.py [hunk 5]
```diff
@@ -356,4 +257,142 @@ def bbox_coords(xs,ys, pad='extends'):
...
+            band = reduced.interp(
+                dim_0=np.arange(src.height),
+                dim_1=np.arange(src.width),
+                method='nearest').values.astype(src.dtypes[0])
+        else:
+            band = src.read(1)
+
+        # if constant is not None:
+        #    band = src.read(1)
+        #    band[band >= 0] = int(constant)
```
Comment (alevieux) [inline src/xsar/utils.py:322]: Removing comment line

**Comment by alevieux** [comment]
For update workflow GitHub action you can add rioxarray dependancies in environment install

for osx 

https://github.com/umr-lops/xsar/blob/26ba7e98b32aa4301469af3d3103d88f6e080c4e/.github/workflows/mac-os-install-test.yml#L33

for ubuntu

https://github.com/umr-lops/xsar/blob/26ba7e98b32aa4301469af3d3103d88f6e080c4e/.github/workflows/ubuntu-install-test.yml#L33

for windows

https://github.com/umr-lops/xsar/blob/26ba7e98b32aa4301469af3d3103d88f6e080c4e/.github/workflows/windows-install-test.yml#L33

File: src/xsar/sentinel1_dataset.py (hunk 1 above)
Comment (oarcher) [reply src/xsar/sentinel1_dataset.py:94]: This check might be turned into a test: can you add a test that check s1meta serialization ?

File: setup.py (hunk 2 above)
Comment (oarcher) [reply setup.py:39]: migration to rioxarray is done, i'm going to try to reactivate this backend

File: src/xsar/sentinel1_dataset.py (hunk 3 above)
Comment (oarcher) [reply src/xsar/sentinel1_dataset.py:139]: ok, some old stuff here. can be removed.

File: src/xsar/utils.py (hunk 4 above)
Comment (oarcher) [reply src/xsar/utils.py]: ok, code cleaned

File: src/xsar/utils.py (hunk 5 above)
Comment (oarcher) [reply src/xsar/utils.py]: ok, code cleaned

File: src/xsar/sentinel1_dataset.py (hunk 1 above)
Comment (alevieux) [reply src/xsar/sentinel1_dataset.py:94]: I add test in my future pull request for purest

Comment (alevieux) [reply src/xsar/sentinel1_dataset.py:94]: What is variable tuple in assertion ?

**Comment by alevieux** [comment]
We can delete the python 3.6 ubuntu test because the new dependencies rioxarray not support python 3.6 
[https://github.com/corteva/rioxarray/issues/215](https://github.com/corteva/rioxarray/issues/215)

In consequence xsar it's available with python>=3.7

File: src/xsar/sentinel1_dataset.py (hunk 1 above)
Comment (oarcher) [reply src/xsar/sentinel1_dataset.py:94]: It's just a test to check that `s1meta.coords2ll(100, 100)` returns a type tuple

**Pull request merged**